FWCore/Framework/interface/ESPreFunctorDecorator.h

```diff
@@ -33,7 +33,11 @@ namespace edm {
     class ESPreFunctorDecorator {
     public:
       ESPreFunctorDecorator(const TFunctor& iCaller) : caller_(iCaller) {}
-      const ESPreFunctorDecorator& operator=(const ESPreFunctorDecorator&) = delete;  // stop default
+      ESPreFunctorDecorator() = delete;
+      ESPreFunctorDecorator(ESPreFunctorDecorator&&) = default;
+      ESPreFunctorDecorator(ESPreFunctorDecorator const&) = default;
+      ESPreFunctorDecorator& operator=(const ESPreFunctorDecorator&) = delete;  // stop default
+      ESPreFunctorDecorator& operator=(ESPreFunctorDecorator&&) = delete;
```

Comment:
Just to double check, this class is copyable and movable via constructor, but not via assignment? I.e. assigning to a `caller_` (when not initializing it) is what we want to avoid? But copying `caller_` as such is fine?

Reply:
The original code stopped the regular op= but the compiler warning complained about the default copy constructor.  To make the code work, I had to require the copy constructor (and figured move constructor would be fine). Then I just picked the move op= to be treated like the regular op=.